src/ray/core_worker/core_worker.cc

```diff
@@ -65,28 +72,35 @@ CoreWorker::CoreWorker(const WorkerType worker_type, const Language language,
   }
 
   // Initialize gcs client.
-  gcs_client_ =
-      std::unique_ptr<gcs::RedisGcsClient>(new gcs::RedisGcsClient(gcs_options));
-  RAY_CHECK_OK(gcs_client_->Connect(io_service_));
+  RAY_CHECK_OK(gcs_client_.Connect(io_service_));
 
   // Initialize profiler.
   profiler_ = std::make_shared<worker::Profiler>(worker_context_, node_ip_address,
                                                  io_service_, gcs_client_);
 
-  object_interface_ = std::unique_ptr<CoreWorkerObjectInterface>(
-      new CoreWorkerObjectInterface(worker_context_, raylet_client_, store_socket,
-                                    use_memory_store, check_signals));
-
   // Initialize task execution.
-  int rpc_server_port = 0;
   if (worker_type_ == WorkerType::WORKER) {
-    task_execution_interface_ = std::unique_ptr<CoreWorkerTaskExecutionInterface>(
-        new CoreWorkerTaskExecutionInterface(*this, worker_context_, raylet_client_,
-                                             *object_interface_, profiler_,
-                                             task_execution_callback));
-    rpc_server_port = task_execution_interface_->worker_server_.GetPort();
+    RAY_CHECK(task_execution_callback_ != nullptr);
+
+    // Initialize task receivers.
+    auto execute_task = std::bind(&CoreWorker::ExecuteTask, this, std::placeholders::_1,
+                                  std::placeholders::_2, std::placeholders::_3);
+    task_receivers_.emplace(
```

Comment:
Can you also get rid of this array and the `CoreWorkerTaskReceiver` class? The interface doesn't seem to have any methods.